src/elements/common/annotator-context/withAnnotations.tsx

```diff
@@ -61,6 +80,25 @@ export default function withAnnotations<P extends object>(
             this.state = { ...defaultState, activeAnnotationId };
         }
 
+        componentDidUpdate(
+            _prevProps: WithAnnotationsProps,
+            { hasAnnotator: prevHasAnnotator, error: prevError }: AnnotatorState,
+        ) {
+            const { hasAnnotator, error } = this.state;
+
+            if (
+                (prevHasAnnotator !== hasAnnotator || prevError !== error) &&
+                hasAnnotator &&
+                this.annotator &&
+                error &&
+                'origin' in error &&
+                error.origin === ORIGIN_ACTIVITY_SIDEBAR
```

Comment:
Maybe this can be simplified if:
- `handleAnnotationsFetchError` doesn't worry about calling `this.escalateAnnotationsFetchError` but relies on `componentDidUpdate`
- we pull out conditions as variables

Something like:
```
const hasAnnotatorChanged = this.annotator && prevHasAnnotator !== hasAnnotator;
const hasErrorChanged = error && prevError !== error;
if (hasAnnotatorChanged && hasErrorChanged) {
  this.escalateAnnotationsFetchError(error);
  this.annotator.emit('annotations_visible_set', false);
}
```